### DISCUSSION TITLE
Let Text component support flex prop

### DISCUSSION THREAD

**Discussion by selbekk** [Shopify/restyle#244] [Q&A]
Is there a way we can make the default Text-component support the `flex` prop?

Today, this doesn't work:

```tsx
<Text flex="1">My text takes up all the space</Text>
```

But this does:

```tsx
<Text style={{ flex: "1" }}>My text takes up all the space</Text>
```

**Comment by flexbox** [comment]
Hey @selbekk 
Thank you for the issue 

Have you read this thread? https://github.com/Shopify/restyle/issues/132#issuecomment-1079873056

> `layout` is not included in `textRestyleFunctions`

For now instead, of using `createText` you can "manually" create your `Text` component to include the props you want.

**Reply by selbekk** [reply]
Hi! Sorry for the lacking response.

I read the thread, but didn't feel any wiser. What is the intent behind not allowing Text content to have flex properties? There are several common use cases for having to specify flex props (flexGrow, flexShrink, flex, alignSelf, justifySelf etc) for text nodes.

I've already created my own Text component, which works fine, but I'd love to discuss changing the default behavior here – or at least understanding the reasoning for why not.
